rllib/algorithms/mbmpo/mbmpo_torch_policy.py

```diff
@@ -1,135 +1,90 @@
-import gym
 from gym.spaces import Box, Discrete
 import logging
 from typing import Tuple, Type
 
 import ray
-from ray.rllib.agents.a3c.a3c_torch_policy import vf_preds_fetches
-from ray.rllib.algorithms.maml.maml_torch_policy import (
-    setup_mixins,
-    maml_loss,
-    maml_stats,
-    maml_optimizer_fn,
-    KLCoeffMixin,
-)
-from ray.rllib.agents.ppo.ppo_tf_policy import setup_config
-from ray.rllib.evaluation.postprocessing import compute_gae_for_sample_batch
+from ray.rllib.algorithms.maml.maml_torch_policy import MAMLTorchPolicy
 from ray.rllib.models.catalog import ModelCatalog
 from ray.rllib.models.modelv2 import ModelV2
 from ray.rllib.models.torch.torch_action_dist import TorchDistributionWrapper
-from ray.rllib.policy.policy import Policy
-from ray.rllib.policy.policy_template import build_policy_class
 from ray.rllib.utils.error import UnsupportedSpaceException
 from ray.rllib.utils.framework import try_import_torch
-from ray.rllib.utils.torch_utils import apply_grad_clipping
-from ray.rllib.utils.typing import TrainerConfigDict
 
 torch, nn = try_import_torch()
 
 logger = logging.getLogger(__name__)
 
 
-def validate_spaces(
-    policy: Policy,
-    observation_space: gym.spaces.Space,
-    action_space: gym.spaces.Space,
-    config: TrainerConfigDict,
-) -> None:
-    """Validates the observation- and action spaces used for the Policy.
-
-    Args:
-        policy (Policy): The policy, whose spaces are being validated.
-        observation_space (gym.spaces.Space): The observation space to
-            validate.
-        action_space (gym.spaces.Space): The action space to validate.
-        config (TrainerConfigDict): The Policy's config dict.
+class MBMPOTorchPolicy(MAMLTorchPolicy):
+    def __init__(self, observation_space, action_space, config):
+        # Validate spaces.
+        # Only support single Box or single Discrete spaces.
+        if not isinstance(action_space, (Box, Discrete)):
+            raise UnsupportedSpaceException(
+                "Action space ({}) of {} is not supported for "
+                "MB-MPO. Must be [Box|Discrete].".format(action_space, self)
+            )
+        # If Box, make sure it's a 1D vector space.
+        elif isinstance(action_space, Box) and len(action_space.shape) > 1:
+            raise UnsupportedSpaceException(
+                "Action space ({}) of {} has multiple dimensions "
+                "{}. ".format(action_space, self, action_space.shape)
+                + "Consider reshaping this into a single dimension Box space "
+                "or using the multi-agent API."
+            )
+
+        config = dict(ray.rllib.algorithms.mbmpo.mbmpo.DEFAULT_CONFIG, **config)
+        super().__init__(observation_space, action_space, config)
+
+    def make_model_and_action_dist(
+        self,
+    ) -> Tuple[ModelV2, Type[TorchDistributionWrapper]]:
```

Comment:
Nite: Don't have to do this in this PR: We should unify the APIs here for torch: `make_model_and_action_dist` and tf: `make_model` (action_dist?? :) ).

Reply:
definitely, we should.